src/validation/difficulty.cairo

```diff
@@ -21,10 +23,37 @@ pub fn validate_bits(target: u256, bits: u32) -> Result<(), ByteArray> {
 /// Adjusts difficulty target given the block height and timestamp.
 /// Returns new difficulty target and new epoch start time.
```

Comment:
Let's add here that previous block time is used for actual timespan calculation and current block time will be used as the new epoch time.

I'm thinking if it makes sense to split this function into two similarly to https://github.com/bitcoin/bitcoin/blob/6d546336e800f7b8990fececab6bc08413f28690/src/pow.cpp#L49 to avoid doing many things and returning a tuple.

Can you please add this as todo for now, maybe will address that in the future

Reply:
I think it makes sense to calculate target and epoch at the same time. If we separate them, we'd still have to use the same conditions.